llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -552,17 +552,17 @@ struct BasicAAResult::DecomposedGEP {
   APInt Offset;
   // Scaled variable (non-constant) indices.
   SmallVector<VariableGEPIndex, 4> VarIndices;
-  // Are all operations inbounds GEPs or non-indexing operations?
+  // Nowrap flags common to all GEP operations involved in expression.
   // (std::nullopt iff expression doesn't involve any geps)
-  std::optional<bool> InBounds;
+  std::optional<GEPNoWrapFlags> NWFlags;
```

Comment:
```suggestion
  GEPNoWrapFlags NWFlags = GEPNoWrapFlags::all();
```
Should make things a bit clearer. I don't think the optional adds anything here...

Reply:
Good point - I've made this initialize with `GEPNoWrapFlags::none()`, with some logic in `DecomposeGEPExpression` so that we initialize on the first GEP we encounter - as this seemed to make sense to avoid ever having DecompGEP2 retain `GEPNoWrapFlags::all()` in the case where the decomposition of `V2` does not encounter any GEPs. 

This doesn't seem to be an issue at present, but seems cleaner to make sure the flags are always correct in this way?

Reply:
No GEP would be equivalent to a zero-offset GEP, for which all of inbounds, nusw, and nuw always hold. In the past, it was unclear whether inbounds held for zero-offset GEPs, but this has been decided now. As such, initializing this to all() is correct, and should make things a bit simpler because we don't need the separate flag.

Reply:
Ah yes, I can see this case is actually spelled out in LangRef, thanks for pointing that out. Updated to initialize with all() and remove the now unnecessary conditional.